From 31277a47c85a65a9489cd913f585cbaa65ed2931 Mon Sep 17 00:00:00 2001
From: svenate <salini.venate@intel.com>
Date: Tue, 15 Mar 2022 23:20:58 +0530
Subject: [PATCH] Remove -fno-strict-overflow to fix Android T build errors

This is a WA patch to fix the below build error in
Android T:

clang: error: argument unused during compilation:
'-fno-strict-overflow' [-Werror,-Wunused-command-line-argument]

Signed-off-by: svenate <salini.venate@intel.com>
---
 Android.mk                      | 1 -
 wpa_supplicant_8_lib/Android.mk | 3 +--
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/Android.mk b/Android.mk
index e753779..0aef1bd 100644
--- a/Android.mk
+++ b/Android.mk
@@ -13,7 +13,6 @@ LOCAL_CPPFLAGS += -Wno-unused-parameter -Wno-int-to-pointer-cast -Wno-missing-fi
 LOCAL_CPPFLAGS += \
         -D_FORTIFY_SOURCE=2 \
         -fstack-protector-strong \
-        -fno-strict-overflow \
         -fno-delete-null-pointer-checks \
         -fwrapv \
         -Wformat -Wformat-security \
diff --git a/wpa_supplicant_8_lib/Android.mk b/wpa_supplicant_8_lib/Android.mk
index 10dfdc1..d79f12b 100644
--- a/wpa_supplicant_8_lib/Android.mk
+++ b/wpa_supplicant_8_lib/Android.mk
@@ -44,8 +44,7 @@ endif
 ifdef CONFIG_ANDROID_LOG
 L_CFLAGS += -DCONFIG_ANDROID_LOG
 endif
-L_CFLAGS += -fno-strict-overflow \
-            -fno-delete-null-pointer-checks \
+L_CFLAGS += -fno-delete-null-pointer-checks \
             -fwrapv
 
 ########################
-- 
2.17.1

